pkg/cmd/pulumi/up.go

```diff
@@ -566,6 +568,9 @@ func newUpCmd() *cobra.Command {
 	cmd.PersistentFlags().BoolVar(
 		&showReplacementSteps, "show-replacement-steps", false,
 		"Show detailed resource replacement creates and deletes instead of a single step")
+	cmd.PersistentFlags().StringVarP(
+		&lockTimeout, "lock-timeout", "l", "",
```

Comment:
I would default this to "0", i.e. don't wait. To maintain the current behavior.

Reply:
Only because it's an error not to include the units. Parse("1") returns `missing unit in duration "1"` and so you get 0s.
A) We should be clear that the default is the zero value of no waiting.
B) We should probably support a default unit of seconds here. If someone does "--lock-timeout=30" the reasonable thing is probably to assume 30 seconds, not error.

Reply:
I have updated this to default to `0s`.

I'm not sure that adding a default unit is a smart move personally and the Terraform version of this feature does not: https://developer.hashicorp.com/terraform/cli/commands/plan#lock-timeout-duration